### ISSUE TITLE
Input doesn't work when running under Cygwin's bash via a PuTTY session

### ISSUE THREAD

**Issue by gene-pavlovsky** [ianxm/ihx#13]
I haven't tried this on an actual Linux system yet, but when running under Cygwin's bash via a PuTTY session), ihx doesn't seem to connect to stdin.
After I entered var a=1; - nothing happened, so I pressed Ctrl+C:

```
# haxelib.exe run ihx
haxe interactive shell v0.3.4
type "help" for help
>> var a=1;


14:07:42 ~
# var a=1;
bash: var: command not found
```

**Comment by ianxm** [comment]
I don't follow. you're using putty to connect to a windows box running cygwin with sshd?

**Comment by gene-pavlovsky** [comment]
PuTTY actually includes contrib/cygtermd, a companion for use with Cygwin (on local PC).
http://www.chiark.greenend.org.uk/~sgtatham/putty/wishlist/cygwin-terminal-window.html
The reason I use PuTTY is it has a nice terminal emulator that works well with various full-screen UNIX console programs (e.g. text editors) and there are nice multi-tabbed wrappers around it.
And apparently using Cygtermd is not important, I also have sshd running (on my local PC), connecting with putty to localhost via ssh and there is same behavior, ihx doesn't read any input.

**Comment by NQNStudios** [comment]
I can reproduce the log above using an Msys2 terminal on my Windows box. Something like it also happens when I try to run ihx from a `comint-mode` terminal in Emacs--minus the part where the command gets sent to Bash after I type ctrl+c.

If I find a fix for this I'll let you know.

**Comment by ianxm** [comment]
you're right that ihx doesn't work from `comint-mode`. I get the same behavior in `shell` and `eshell` also, but `term` is ok.

**Comment by NQNStudios** [comment]
Just out of curiosity, do you ever write Haxe code in Emacs? I tried a
bunch of the different outdated haxe modes and none of them could get
formatting right.

**Comment by ianxm** [comment]
I write everything in emacs. I use the `haxe-mode` melpa package. it isn't perfect but is good enough.

**Comment by NQNStudios** [comment]
Do you disable electric indentation mode? I found that it was always
breaking my indentation as I typed. But I don't know a better alternative
for making indentation automatic.
